spec/dummy/app/resources/article_resource.rb

```diff
@@ -24,4 +24,8 @@ def id=(external_id)
   # Setting this attribute is an easy way to test that authorizations work even
   # when the model has validation errors
   attributes :blank_value
+
+  def self.creatable_fields(context)
+    super + [:id]
+  end
```

Comment:
Ok turns out that these lines are indeed needed since v0.8.3 of JR as IDs are generated on the client side.

See https://github.com/cerebris/jsonapi-resources/releases/tag/v0.8.3